app/api/search/specs/search.spec.js

```diff
@@ -85,16 +86,27 @@ describe('search', () => {
         .catch(catchErrors(done));
     });
 
-    it('should perform the search on unpublished documents also', done => {
+    it('should include unpublished documents if logged in', done => {
+      const user = { _id: 'userId' };
       search
-        .searchSnippets('unpublished', 'unpublishedSharedId', 'en')
+        .searchSnippets('unpublished', 'unpublishedSharedId', 'en', user)
         .then(snippets => {
           expect(snippets.fullText.length).toBe(1);
           done();
         })
         .catch(catchErrors(done));
     });
 
+    it('should not include unpublished if not logged in', done => {
+      search
+        .searchSnippets('unpublished', 'unpublishedSharedId', 'en', undefined)
+        .then(snippets => {
+          expect(snippets.fullText.length).toBe(0);
+          done();
+        })
+        .catch(catchErrors(done));
+    });
+
```

Comment:
Two observations here:
- please do not create new tests with the `done` flow, use async (and perhaps refactor the previous test?).  There is no longer need for the catchErrors, etc. It's much cleaner.
- I think it's dangerous to have an all-or-nothing test.  The first test tests that all documents come through, the second test that none come through.  If there were a coding bug that prevented any result to come through, test two would still pass green.  So please, make it so that test one has TWO results and test two yields ONE result so that we know that: A. Some results are coming, B. Partial results are coming through.

Reply:
Well, for the record, I'm a complete moron: this endpoint is intended for single documents.  Disregard the second part of the comment.